tests/index.test.ts

```diff
@@ -640,8 +679,424 @@ describe("OpenFGA SDK", function () {
+            tuple_key: tupleKey,
+          },
+          expect.objectContaining({ Authorization: "Bearer test-token" })
+        )
+        .reply(429, {
+          code: "rate_limit_exceeded",
+          message: "rate limited",
+        });
+
+      // Second request succeeds
+      nock(basePath)
+        .post(
+          `/stores/${storeId}/check`,
+          {
+            tuple_key: tupleKey,
+          },
+          expect.objectContaining({ Authorization: "Bearer test-token" })
+        )
+        .reply(200, { allowed: true });
+
+      const result = await fgaApi.check(baseConfig.storeId!, { tuple_key: tupleKey });
+      expect(result.allowed).toBe(true);
+    });
+
+    it("should handle invalid Retry-After header format", async () => {
+      // First request fails with 429, invalid Retry-After header
+      nock(basePath)
+        .post(
+          `/stores/${storeId}/check`,
+          {
+            tuple_key: tupleKey,
+          },
+          expect.objectContaining({ Authorization: "Bearer test-token" })
+        )
+        .reply(429, {
+          code: "rate_limit_exceeded",
+          message: "rate limited",
+        }, {
+          "Retry-After": "not-a-number"
+        });
+
+      // Second request succeeds
+      nock(basePath)
+        .post(
+          `/stores/${storeId}/check`,
+          {
+            tuple_key: tupleKey,
+          },
+          expect.objectContaining({ Authorization: "Bearer test-token" })
+        )
+        .reply(200, { allowed: true });
+
+      const result = await fgaApi.check(baseConfig.storeId!, { tuple_key: tupleKey });
+      expect(result.allowed).toBe(true);
+    });
+
+    it("should cap retry time when Retry-After exceeds max time", async () => {
+      // First request fails with 429, Retry-After exceeds max wait time (1000ms configured)
+      nock(basePath)
+        .post(
+          `/stores/${storeId}/check`,
+          {
+            tuple_key: tupleKey,
+          },
+          expect.objectContaining({ Authorization: "Bearer test-token" })
+        )
+        .reply(429, {
+          code: "rate_limit_exceeded",
+          message: "rate limited",
+        }, {
+          "Retry-After": "10" // 10 seconds, exceeds maxWaitTime of 1000ms
+        });
+
+      // Second request succeeds
+      nock(basePath)
+        .post(
+          `/stores/${storeId}/check`,
+          {
+            tuple_key: tupleKey,
+          },
+          expect.objectContaining({ Authorization: "Bearer test-token" })
+        )
+        .reply(200, { allowed: true });
+
+      const startTime = Date.now();
+      const result = await fgaApi.check(baseConfig.storeId!, { tuple_key: tupleKey });
+      const elapsedTime = Date.now() - startTime;
+
+      expect(result.allowed).toBe(true);
+      // Should wait close to maxWaitTime (1000ms) but not 10 seconds
+      expect(elapsedTime).toBeLessThan(5000);
+    });
+
+    it("should retry immediately when Retry-After is 0", async () => {
```

Comment:
This should not be the case, as `isValidRetryDelay` should be false when retry after is <1 and it should default to min wait